### ISSUE TITLE
AssertionError on stderr when calling gevent.wait() on an already-set Event with count != None

### ISSUE THREAD

**Issue by ekimekim** [gevent/gevent#1540]
* gevent version: 1.4.0
* Python version: CPython 2.7.14 downloaded from archlinux official repo, package python2 2.7.17-1
* Operating System: Arch linux, kernel 5.4.6-arch3-1

### Description:

When attempting to wait upon a list of `gevent.event.Event` objects using `gevent.wait()`, where at least one of those events is already set, and the `count` arg to `gevent.wait()` is not `None` (eg. is `1`), then the following is printed to stderr:
```
AssertionError
2020-03-06T03:12:17Z <callback at 0x7fc078b74280 stopped> failed with AssertionError
```
(obviously the timestamp and address vary)

The `gevent.wait()` function still returns correctly, this appears to be entirely a cosmetic issue. Given how often such calls may occur however, it is still a problem.

### What I've run:

```pycon
Python 2.7.17 (default, Oct 22 2019, 09:14:09) 
[GCC 9.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import gevent.event
>>> e = gevent.event.Event()
>>> e.set()
>>> gevent.wait([e]) # succeeds
[<gevent._event.Event object at 0x7ffae2dd24b0>]
>>> gevent.wait([e], count=1) # prints error but still succeeds
AssertionError
2020-03-06T03:20:45Z <callback at 0x7ffae24e9eb0 stopped> failed with AssertionError

[<gevent._event.Event object at 0x7ffae2dd24b0>]
```

I did some diving before finding the triggering conditions, and I found that the assertion error seems to have a hidden traceback (output is from `traceback.extract_tb()`):
```
[('src/gevent/_abstract_linkable.py', 114, 'gevent.__abstract_linkable.AbstractLinkable._notify_links', None), ('src/gevent/_abstract_linkable.py', 148, 'gevent.__abstract_linkable.AbstractLinkable._notify_links', None)]
```
which corresponds to this assert failure in `_abstract_linkable.py`:
```
assert self._notifier is notifier
```

**Comment by jamadden** [comment]
Thanks for the report! This is still an issue in current master.

**Comment by karlb** [comment]
Does this cause any problems except printing the unexpected messages? The issue description say it "appears to be entirely a cosmetic issue", but I expect that a failing `assert` will prevent the remainder of the function from getting executed.

**Comment by ekimekim** [comment]
In practice we haven't hit this issue much in running code - I only noticed it due to a bug in my code that meant the Event I was waiting on would always be set. So as I said, it _appears_ to be entirely cosmetic. There may be further issues down the line if it happens a lot, but the observed behaviour of the `gevent.wait()` function seems to be unchanged.

**Comment by jamadden** [comment]
Fixed in #1613
